src/components/legend/helpers.js

```diff
@@ -0,0 +1,25 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+var Plots = require('../../plots/plots');
+
+
+exports.legendGetsTrace = function legendGetsTrace(trace) {
+    return trace.visible && Plots.traceIs(trace, 'showLegend');
+};
+
+exports.isGrouped = function isGrouped(legendLayout) {
+    return (legendLayout.traceorder || '').indexOf('grouped') !== -1;
+};
+
+exports.isReversed = function isReversed(legendLayout) {
```

Comment:
Yep. I'm happy with helper files for utils local only to a specific module, and `var Helpers = require('./helpers');` but just because in my mind, the Capital letter signifies that it's not a variable or consumable (e.g. `attributes`).